Cabal/Distribution/Simple/GHC/Internal.hs

```diff
@@ -182,8 +193,11 @@ configureToolchain _implInfo ghcProg ghcInfo =
              withTempFile tempDir ".o" $ \testofile testohnd -> do
                hPutStrLn testchnd "int foo() { return 0; }"
                hClose testchnd; hClose testohnd
-               runProgram verbosity ghcProg ["-c", testcfile,
-                                             "-o", testofile]
+               runProgram verbosity ghcProg
+                          [ "-hide-all-packages"
```

Comment:
Right, so the ghc behaviour with these environment files is that it uses them by default unless you're using `-hide-all-packages` to get a clean environment. There's a couple places in Cabal where we invoke ghc where we don't want to pick up any local environment. Most of the time doing so would be benign, but if your local environment file is borked then we still want to work, by ignoring it. So that's what these few patches do.

A test would look like making a .ghc.env file that has bogus entries and then use Cabal to do things. That's essentially the situation that provoked me to make these fixes.